decompiler/structures/pseudo/z3_logic.py

```diff
@@ -42,13 +43,26 @@ class Z3Converter(BaseConverter):
 
     def __init__(self):
         self._context = Context()
+        self.OPERATIONS_INVALID_BOOLREF_OP = {
```

Comment:
Can you not have this in the `init`, but below where the other Operation-dictionaries are (end of file)

Reply:
I see the problem now, and that is a good question, what also would work, but not sure whether it makes it better is:
```python
    OPERATIONS_INVALID_BOOLREF_OP = {
        OperationType.minus: operator.sub,
        OperationType.plus: operator.add,
        OperationType.multiply: operator.mul,
        OperationType.divide: operator.truediv,
        OperationType.left_shift: operator.lshift,
        OperationType.right_shift: operator.rshift,
        OperationType.less: operator.lt,
        OperationType.less_or_equal: operator.le,
        OperationType.greater: operator.gt,
        OperationType.greater_or_equal: operator.ge,
        OperationType.modulo: operator.mod,
    }
```
and
```python
converter = lambda a, b: self._convert_invalid_boolref_op(
                a, b, self.OPERATIONS_INVALID_BOOLREF_OP.get(operation.operation, None)
            )
```
another option could be to exclude the function `_convert_invalid_boolref_op` from the class:
```python
    def _convert_invalid_boolref_op(a: BoolRef, b: BoolRef, op: Operation) -> BitVecRef:
        return op(
            If(a, BitVecVal(1, 1, ctx=a.ctx), BitVecVal(0, 1, ctx=a.ctx), ctx=a.ctx),
            If(b, BitVecVal(1, 1, ctx=b.ctx), BitVecVal(0, 1, ctx=b.ctx), ctx=b.ctx),
        )
```

Reply:
A bit ugly, but way better. Will change to it.